sfs/mono/source.py

```diff
@@ -189,31 +187,30 @@ def point_modal(omega, x0, n0, grid, L, N=None, deltan=0, c=None):
     x0 = util.asarray_1d(x0)
     x, y, z = util.as_xyz_components(grid)
 
+    if np.isscalar(N):
+        N = N * np.ones(3, dtype=int)
+
     if N is None:
-        # determine maximum modal order per dimension
-        Nx = int(np.ceil(L[0]/np.pi * k))
-        Ny = int(np.ceil(L[1]/np.pi * k))
-        Nz = int(np.ceil(L[2]/np.pi * k))
-        mm = range(Nx)
-        nn = range(Ny)
-        ll = range(Nz)
-    elif np.isscalar(N):
-        # compute up to a given order
-        mm = range(N)
-        nn = range(N)
-        ll = range(N)
-    else:
-        # compute field for one order combination only
-        mm = [N[0]]
-        nn = [N[1]]
-        ll = [N[2]]
+            N = [None, None, None]
+
+    orders = [0, 0, 0]
+    for i in range(3):
+        if N[i] is None:
+            # compute max order
+            orders[i] = range(np.ceil(L[i]/np.pi * k) + 1)
```

Comment:
You should keep the conversion to `int` here, because `np.ceil()` returns a `float`.